### PR TITLE
Deprecate public use of Formatter.pprint_val.

### PR DESCRIPTION
**Pull request by anntzer**
It's really a helper method.

## PR Checklist

- [ ] Has Pytest style unit tests
- [ ] Code is [Flake 8](http://flake8.pycqa.org/en/latest/) compliant
- [ ] New features are documented, with examples if plot related
- [ ] Documentation is sphinx and numpydoc compliant
- [ ] Added an entry to doc/users/next_whats_new/ if major new feature (follow instructions in README.rst there)
- [ ] Documented in doc/api/api_changes.rst if API changed in a backward-incompatible way

### CODE REVIEW

File: lib/matplotlib/ticker.py [hunk 1]
```diff
@@ -294,6 +294,13 @@ def _set_locator(self, locator):
         """ Subclasses may want to override this to set a locator. """
         pass
 
+    @cbook.deprecated("3.1")
+    def pprint_val(self, *args, **kwargs):
+        return self._pprint_val(*args, **kwargs)
+
+    def _pprint_val(self, *args, **kwargs):
+        raise NotImplementedError
```
Comment (timhoffm) [inline lib/matplotlib/ticker.py]: Should get a comment, what this is supposd to do and how subclasses should implement it. This info is targeted at other developers.

Comment (anntzer) [reply lib/matplotlib/ticker.py]: Moved the deprecation to each individual class, so no need for the NotImplemented base class version anymore.

**Comment by tacaswell** [comment]
Thanks @anntzer .

This is one we should be willing to roll back if we get complaints.

**Comment by anntzer** [comment]
@tacaswell Just to clarify why I deprecated these (it's not just "oh let's make them private because we can"): Right now it's quite difficult to follow the code flow in ticker code, and these pprint_val methods do not help at all as they can be inherited (so it's not even immediately clear which one you're getting) and don't have consistent semantics either.  So the plan is to inline the relevant parts of the code into the callers as much as possible, to avoid having to trace back which pprint_val is being used where.

**Pull request merged**